k3sAndProviderServices.sh

```diff
@@ -153,6 +153,18 @@ update_coredns_config() {
 
 # Add control plane node logic
 if [[ "$mode" == "init" ]]; then
+    # Ensure jq is installed for JSON processing
+    if ! command -v jq &> /dev/null; then
+        echo "jq is not installed. Installing jq..."
+        apt-get update && apt-get install -y jq
+    fi
+
+    # Ensure yq is installed for YAML processing
+    if ! command -v yq &> /dev/null; then
+        echo "yq is not installed. Installing yq..."
+        sudo curl -L https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -o /usr/bin/yq && chmod +x /usr/bin/yq
```

Comment:
Please remove "sudo".
The script is supposed to be run as root anyway.

Reply:
Done. Thanks for pointing it out.